core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala

```diff
@@ -71,6 +71,13 @@ private[spark] object PythonUtils {
     vs.toArray().asInstanceOf[Array[T]]
   }
 
+  /**
+   * Convert list of java.lang.Double into array of Double
+   */
+  def doubleJListToArray(vs: JList[java.lang.Double]): Array[Double] = {
+    vs.asScala.map(_.doubleValue).toArray
```

Comment:
I don't recall any methods that can actually handle `Array[scala.Double]` (`Array[java.lang.Double]` wouldn't be an issue, and normally things are handled through `Seqs`).

Reply:
Oh, we don't need this. We can manually create an array of double in PySpark side. For example as below:

```python
# Create an array of objects
a_list = [...]
double_class = self._spark_session._jvm.Double
arr = SparkContext._gateway.new_array(str_class, len(a_list))
for i in range(len(a_list)):
    arr[i] = a_list[i]
```

Reply:
Are you sure about it? Type erasure, on which we depend all over the place, doesn't apply to `Arrays`, so I fairly confident this won't do.

I will double check that of course.

Reply:
OK. As expected we cannot use `_jvm.Double`, because there is not type erasure or implicit conversion in place:

> Method percentile_approx([class org.apache.spark.sql.Column, class [Ljava.lang.Double;, class java.lang.Integer]) does not exist

and we cannot use `_jvm.scala.Double` because:

> Cannot convert java.lang.Double to scala.Double